pkg/lib/authenticationflow/declarative/intent_create_authenticator_password.go

```diff
@@ -81,6 +81,9 @@ func (i *IntentCreateAuthenticatorPassword) ReactTo(ctx context.Context, deps *a
 			if err != nil {
 				return nil, err
 			}
+			if !IsBotProtectionSpecialErrorSuccess(bpSpecialErr) {
+				return nil, bpSpecialErr
+			}
```

Comment:
Instead of adding IsBotProtectionSpecialErrorSuccess, maybe we should change how HandleBotProtection behave.

1. It returns `(nil, err)`. So that we will `if err != nil { return nil, err }` early. Note that err COULD be specialErr.
2. It returns `(specialErr, nil)`. So that the special error is returned along with the normal return value.

What do you think?

Reply:
I think this ~10-12 lines of re-appearing code should be refactored as a single util function, which has your mentioned return interface 🙏

Reply:
On second thought, I just moved `IsBotProtectionSpecialErrorSuccess` checking into `HandleBotProtection` as you suggested.

### Rationale
IMO Checking `isBPRequired` and actually performing `verifyBP` should be 2 separate function calls.